Source: https://github.com/gentoo/gentoo/blob/master/dev-qt/qtwebengine/files/qtwebengine-6.8.1-aarch64-xnnpack.patch

Patch status: *seems* fixed in Qt 6.9+

Temporary (lame) workaround for [1] given backporting[2] (which
is believed to be the fix) does not seem trivial. A better fix
is welcome if have one and can test it, but otherwise this can
likely be removed in Qt 6.9.0.

[1] https://bugs.gentoo.org/945843
[2] https://github.com/google/XNNPACK/pull/1789
--- a/src/3rdparty/chromium/third_party/xnnpack/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/xnnpack/BUILD.gn
@@ -27,4 +27,5 @@
 
+  cflags = [ "-Wno-error=incompatible-pointer-types" ]
   if (is_clang || !is_win) {
-    cflags = [
+    cflags += [
       "-Wno-unused-function",
